tests/integration_tests/functional/test_mmds.py

```diff
@@ -89,7 +112,18 @@ def test_custom_ipv4(test_microvm_with_api, network_config):
     _, stdout, stderr = ssh_connection.execute_command(cmd)
     _assert_out(stdout, stderr, '')
 
-    pre = 'curl -s -H "Accept: application/json" http://169.254.169.250/'
+    # Generate token.
+    token = _generate_mmds_session_token(
+        ssh_connection,
+        ipv4_address="169.254.169.250",
+        token_ttl=60
+    )
+
+    pre = 'curl -m 2 -s'
+    pre += ' -X GET'
+    pre += ' -H  "X-metadata-token: {}"'.format(token)
+    pre += ' -H "Accept: application/json"'
+    pre += ' http://169.254.169.250/'
```

Comment:
I see this repeated throughout this file. Can we have a function for this so we reuse the code?